pkg/networkmanager/firewall.jsx

```diff
@@ -0,0 +1,265 @@
+                                    <input id={s.id}
+                                           type="checkbox"
+                                           checked={this.state.selected.has(s.id)}
+                                           onClick={this.onToggleService} />
+                                    &nbsp;
+                                    <span>{s.name}</span>
+                                </label>
+                            </li>
+                        ))
+                    }
+                </ul>
+            </div>
+        );
+    }
+}
+
+export class Firewall extends React.Component {
+    constructor() {
+        super();
+
+        this.state = {
+            installed: firewall.installed,
+            enabled: firewall.enabled,
+            services: firewall.services,
+            enabledServices: firewall.enabledServices
+        };
+
+        this.onFirewallChanged = this.onFirewallChanged.bind(this);
+        this.onAddServices = this.onAddServices.bind(this);
+        this.onRemoveService = this.onRemoveService.bind(this);
+    }
+
+    onFirewallChanged() {
+        this.setState({
+            installed: firewall.installed,
+            enabled: firewall.enabled,
+            services: firewall.services,
+            enabledServices: firewall.enabledServices
+        });
+    }
+
+    onSwitchChanged(value) {
+        if (value)
+            firewall.enable();
+        else
+            firewall.disable();
+    }
+
+    onAddServices() {
+        var services = [...this.state.services].map(id => this.state.services[id]);
+        services.sort((a, b) => a.name.localeCompare(b.name));
+
+        var selected = new Set();
+
+        show_modal_dialog(
+            {
+                title: _("Add Services"),
+                body: <AddServicesDialogBody selectionChanged={s => selected = [...s]} />
+            },
+            {
+                cancel_caption: _("Cancel"),
+                actions: [
+                    {
+                        caption: _("Add Services"),
+                        style: 'primary',
+                        clicked: () => firewall.addServices(selected)
+                    }
+                ]
+            });
+    }
+
+    onRemoveService(service) {
+        firewall.removeService(service);
+    }
+
+    componentDidMount() {
+        firewall.addEventListener("changed", this.onFirewallChanged);
+    }
+
+    componentWillUnmount() {
+        firewall.removeEventListener("changed", this.onFirewallChanged);
+    }
+
+    render() {
+        function go_up(event) {
+            if (!event || event.button !== 0)
+                return;
+
+            cockpit.jump("/network", cockpit.transport.host);
+        }
+
+        if (!this.state.installed) {
+            return (
+                <EmptyState title={_("Firewall is not available")} icon="fa fa-exclamation-circle">
+                    <p translatable="yes">Please install the <tt>firewalld</tt> package.</p>
```

Comment:
Can you please update the .pot and make sure that this actually works? I thought the `translate` (aka `translatable`) tags were only considered in .html files. If it doesn't, use `cockpit.format(_("Please ... {0} package"), "firewalld")` here.